docs/ecosystem/kaldi/RunningDistributedThchs30KaldiJobs.md

```diff
@@ -0,0 +1,678 @@
+${THCHS30_PATH}/sge/passwd:/etc/passwd:rw,\
+${THCHS30_PATH}/sge/group:/etc/group:rw,\
+${THCHS30_PATH}/sge:$SGE_CFG_PATH,\
+${THCHS30_PATH}/data:/opt/kaldi/egs/thchs30,\
+${THCHS30_PATH}/mul/s5:/opt/kaldi/egs/mul-thchs30/s5 \
+--input_path /opt/kaldi/egs/thchs30/data \
+--docker_image hadoopsubmarine/kaldi-latest-gpu-base:0.0.1 \
+--num_workers $WORKER_NUM \
+--worker_resources memory=64G,vcores=32,gpu=1 \
+--worker_launch_cmd "sudo mkdir -p /opt/kaldi/egs/mul-thchs30/s5 && \
+sudo cp /opt/kaldi/egs/thchs30/s5/* /opt/kaldi/egs/mul-thchs30/s5 -r && \
+cluster_user=`whoami` domain_suffix="ml.com" && \
+cd /cfg && bash sge_run.sh $WORKER_NUM $SGE_CFG_PATH && \
+if [ $(echo $HOST_NAME |grep "^master-") ] then sleep 2m && cd /opt/kaldi/egs/mul-thchs30/s5 && ./run.sh fi" \
+--verbose
+```
+
+Explanations:
+
+- `>1` num_workers indicates it is a distributed training.
+- Parameters / resources / Docker image of parameter server can be specified separately. For many cases, parameter server doesn't require GPU.We don't need parameter server here
+
+For the meaning of the individual parameters, see the [QuickStart](../../helper/QuickStart.md) page!
+
+*Outputs of distributed training*
+
+Sample output of master:
+```
+...
+Reading package lists...
+Building dependency tree...
+Reading state information...
+The following additional packages will be installed:
+  bsd-mailx cpio gridengine-common ifupdown iproute2 isc-dhcp-client
+  isc-dhcp-common libatm1 libdns-export162 libisc-export160 liblockfile-bin
+  liblockfile1 libmnl0 libxmuu1 libxtables11 ncurses-term netbase
+  openssh-client openssh-server openssh-sftp-server postfix python3-chardet
+  python3-pkg-resources python3-requests python3-six python3-urllib3
+  ssh-import-id ssl-cert tcsh xauth
+Suggested packages:
+  libarchive1 gridengine-qmon ppp rdnssd iproute2-doc resolvconf avahi-autoipd
+  isc-dhcp-client-ddns apparmor ssh-askpass libpam-ssh keychain monkeysphere
+  rssh molly-guard ufw procmail postfix-mysql postfix-pgsql postfix-ldap
+  postfix-pcre sasl2-bin libsasl2-modules dovecot-common postfix-cdb
+  postfix-doc python3-setuptools python3-ndg-httpsclient python3-openssl
+  python3-pyasn1 openssl-blacklist
+The following NEW packages will be installed:
+  bsd-mailx cpio gridengine-client gridengine-common gridengine-exec
+  gridengine-master ifupdown iproute2 isc-dhcp-client isc-dhcp-common libatm1
+  libdns-export162 libisc-export160 liblockfile-bin liblockfile1 libmnl0
+  libxmuu1 libxtables11 ncurses-term netbase openssh-client openssh-server
+  openssh-sftp-server postfix python3-chardet python3-pkg-resources
+  python3-requests python3-six python3-urllib3 ssh-import-id ssl-cert tcsh
+  xauth
+0 upgraded, 33 newly installed, 0 to remove and 30 not upgraded.
+Need to get 12.1 MB of archives.
+After this operation, 65.8 MB of additional disk space will be used.
+Get:1 http://mirrors.aliyun.com/ubuntu xenial/main amd64 libatm1 amd64 1:2.5.1-1.5 [24.2 kB]
```

Comment:
Can replace the source?

Reply:
YES,The "archive.ubuntu.com" source was not added to the dockerfile before. This is the information in the log, and I will replace the content in the log